src/foam/nanos/notification/push/PushRegistryService.js

```diff
@@ -12,25 +12,47 @@ foam.CLASS({
   javaImports: [
     'foam.dao.DAO',
     'foam.nanos.auth.Subject',
-    'foam.nanos.auth.User'
+    'foam.nanos.auth.User',
+    'foam.nanos.auth.AuthService',
+    'foam.nanos.notification.push.iOSNativePushRegistration'
   ],
 
   methods: [
     {
       name: 'subscribe',
 //      type: 'Void',
-//      args: 'Context x, String sub, String endpoint, String key, String auth'
+//      args: 'Context x, String sub, String endpoint, String key, String auth, String token'
       javaCode: `
         User user = ((Subject) x.get("subject")).getUser();
 
         if ( user == null ) throw new IllegalArgumentException("Missing user.");
+        PushRegistration r;
+        if ( token == null ) {
```

Comment:
`SafetyUtil.isEmpty(token)` would check both null and empty string if we could use it here